harness/determined/common/experimental/determined.py

```diff
@@ -55,6 +55,22 @@ def __init__(
         else:
             self._token = None
 
+    @classmethod
+    def _from_session(cls, session: api.Session) -> "Determined":
+        """Create a new Determined object that uses an existing session.
+
+        This constructor exists to help the CLI transition to using SDK methods, most of which are
+        derived from a Determined object at some point in their lifespan.
+
+        WARNING: Determined objects created with this contsructor will not have a token, and so
+        have no access to the oauth API.
+        """
+        new_det = cls.__new__(cls)
```

Comment:
this makes me wonder if we should just accept `session` as a parameter in the default `Determined` constructor. i mean, it'd be weird since it'd be user-facing and i don't see a reason why users would need to pass that in (at least today, maybe in the future users could meaningfully construct a session and pass that in). but i really don't like having two completely independent constructors here. 

eh, i dunno, keep going back and forth in my head on whether it's cleaner for the constructor logic to be in one place (with an if/else) or completely independent.

Reply:
> but i really don't like having two completely independent constructors here.

Totally agree. I believe that this is the current constructor's fault. I think it should not be doing work that a caller might not want (in this case, stuff with sessions and tokens). `client` has a `login` that does this initialization stuff. I think `Determined` should probably have something similar.

More to think about when/if we eventually reconcile `client` and `Determined`.

> if we should just accept session as a parameter

Ryan's got a strong opinion here that `Session` should have been kept private. I haven't thought about it too much. But having it a parameter seems to me like a nice separation of responsibility.